csr_check.go

```diff
@@ -428,11 +428,25 @@ func getServingCert(nodes corev1client.NodeInterface, nodeName string, ca *x509.
 	return cert, nil
 }
 
-// nodeInternalIP returns the first internal IP for the node.
+// nodeInternalIP returns the first valid internal IP for the node as a string
+// suitable for passing to a net.Dialer.
 func nodeInternalIP(node *corev1.Node) (string, error) {
 	for _, address := range node.Status.Addresses {
 		if address.Type == corev1.NodeInternalIP {
-			return address.Address, nil
+			ip := net.ParseIP(address.Address)
+			if ip == nil {
+				klog.Warningf("Node %s has invalid IP address: %q",
+					node.GetName(), address.Address)
+
+				continue
+			}
+
+			// IPv6 addresses must be enclosed in square brackets.
+			if ip.To4() == nil {
+				return fmt.Sprintf("[%s]", ip.String()), nil
+			}
+
+			return ip.String(), nil
```

Comment:
Instead of doing all this, you can just change line 410 from:
```
	kubelet := fmt.Sprintf("%s:%d", host, port)
```

to something like:
```
kubelet := net.JoinHostPort(host, port)
```

except you have to convert port to a string first, i think

Reply:
True. Fixed.